frappe/core/doctype/doctype/test_doctype.py

```diff
@@ -773,6 +773,30 @@ def test_not_in_list_view_for_not_allowed_mandatory_field(self):
 		self.assertTrue(doctype.fields[1].in_list_view)
 		frappe.delete_doc("DocType", doctype.name)
 
+	def test_single_doctype_defaults(self):
+		single = new_doctype(is_single=True).insert()
+
+		# init single
+		frappe.new_doc(single.name).save()
+
+		new_fields = (
+			("new_data_field", "Data", frappe.generate_hash()),
+			("new_int_field", "Int", 42),
```

Comment:
This fails for some reason, get_doc is returning string instead of int. Could be some other bug or caching. :woozy_face: